self-paced-labs/vertex-ai/vertex-ai-qwikstart/lab_exercise.ipynb

```diff
@@ -1567,158 +1717,384 @@
+    "    parser.add_argument('--training-data-uri', dest='training-data-uri',\n",
+    "                        default=os.environ['AIP_TRAINING_DATA_URI'], type=str, help='Training data GCS or BQ URI set during Vertex AI training.')\n",
+    "    parser.add_argument('--validation-data-uri', dest='validation-data-uri',\n",
+    "                        default=os.environ['AIP_VALIDATION_DATA_URI'], type=str, help='Validation data GCS or BQ URI set during Vertex AI training.')\n",
+    "    parser.add_argument('--test-data-uri', dest='test-data-uri',\n",
+    "                        default=os.environ['AIP_TEST_DATA_URI'], type=str, help='Test data GCS or BQ URI set during Vertex AI training.')\n",
+    "    # Model training args.\n",
+    "    parser.add_argument('--learning-rate', dest='learning-rate', default=0.001, type=float, help='Learning rate for optimizer.')\n",
+    "    parser.add_argument('--dropout', dest='dropout', default=0.2, type=float, help='Float percentage of DNN nodes [0,1] to drop for regularization.')    \n",
+    "    parser.add_argument('--batch-size', dest='batch-size', default=16, type=int, help='Number of examples during each training iteration.')    \n",
+    "    parser.add_argument('--n-train-examples', dest='n-train-examples', default=2638, type=int, help='Number of examples to train on.')\n",
+    "    parser.add_argument('--stop-point', dest='stop-point', default=10, type=int, help='Number of passes through the dataset during training to achieve convergence.')\n",
+    "    parser.add_argument('--n-checkpoints', dest='n-checkpoints', default=10, type=int, help='Number of model checkpoints to save during training.')\n",
+    "    \n",
+    "    args = parser.parse_args()\n",
+    "    hparams = args.__dict__\n",
     "\n",
-    "# \n",
-    "earlystopping_callback = tf.keras.callbacks.EarlyStopping(patience=2)\n",
+    "    model.train_evaluate_explain_model(hparams)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86d72c0a",
+   "metadata": {},
+   "source": [
+    "### 3. Write a `Dockerfile` for your custom ML model container"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "75e8f8af",
+   "metadata": {},
+   "source": [
+    "Third, you will write a `Dockerfile` that contains your model code as well as specifies your model code's dependencies.\n",
     "\n",
-    "# Ensure your training job's resilience to VM restarts.\n",
-    "checkpoint_callback = tf.keras.callbacks.ModelCheckpoint(\n",
-    "    filepath=os.environ[\"AIP_MODEL_DIR\"] + '/checkpoints',\n",
-    "    save_weights_only=True,\n",
-    "    monitor='val_loss',\n",
-    "    mode='min')\n",
+    "Notice the base image below is a Vertex AI [pre-built container](https://cloud.google.com/vertex-ai/docs/training/pre-built-containers). This image includes dependencies you need for your training code such TensorFlow and BigQuery SDK which are already tested and built to save you time when building your custom container."
```

Comment:
The one below isn't a Vertex pre-built container, it is a Deep Learning Container.

I am not sure if Vertex pre-built containers are even supposed to work with custom container methods.

Reply:
Thanks, updating. Sloppiness/sleepiness on my part, will update to say its using DL container.

Reply:
Oh I know they are confusing -- and I am sure that the Vertex containers are closely built on the DLC but not the same.

Similar to how I learned that a Vertex Notebook based on a TF2.5 DLC is very similar to, but not at all identical with a TF2.5 DLVM.